scripts/maint/pre-push.git-hook

```diff
@@ -44,8 +44,9 @@ do
 			range="$remote_sha..$local_sha"
 		fi
 
-                if ref_is_upstream_branch "$local_ref" == 0 ||
-                        ref_is_upstream_branch "$remote_ref"  == 0
+                if (ref_is_upstream_branch "$local_ref" == 0 ||
+                        ref_is_upstream_branch "$remote_ref"  == 0) &&
+                        "$local_ref" ne "$remote_ref"
```

Comment:
Does this part need to be in [square brackets] ?

Reply:
Shellcheck does not complain about this, so I think not? Then again I'm really not sure.

Reply:
On second thought, let's use `[...]` for consistency with previous code. See e28bbc8bd5847aba774b5ea38e7e98fbe5dd11fd.